x-pack/plugins/maps/public/classes/sources/es_search_source/es_search_source.tsx

```diff
@@ -174,7 +213,7 @@ export class ESSearchSource extends AbstractESSource {
         label: i18n.translate('xpack.maps.source.esSearch.geoFieldLabel', {
           defaultMessage: 'Geospatial field',
         }),
-        value: this._descriptor.geoField,
+        value: typeof this._descriptor.geoField === 'string' ? this._descriptor.geoField : '',
```

Comment:
If geoField is not provided then an error is thrown in constructor so all of this check logic is not needed.